### PR TITLE
Added one more cave mapgen variation

### PR DESCRIPTION
**Pull request by Night-Pryanik**
#### Summary
SUMMARY: Content "Added one more cave mapgen variation."

#### Purpose of change
More easter eggs!

#### Describe the solution
Added a `dreadful rabbit` monster with very high speed and dodge and tremendous fighting skills.
Added a kite shield from Medieval Stuff mod and made it spawn in cave vicinity. There are also remains of three unfortunate knights with medieval gear and weapons.
The cave itself has a very low chance to spawn as it's an easter egg.
Added a mininuke to the cave's finale - this should resemble the holy hand grenade of Antioch. Only this time it's a reward for defeating the rabbit, not the weapon to kill it. 

#### Describe alternatives you've considered
None.

#### Testing
Debug-spawned this variation to check if everything is ok.

#### Additional context
изображение

### CODE REVIEW

File: data/json/items/armor/misc.json [hunk 1]
```diff
@@ -267,5 +267,27 @@
     "material_thickness": 1,
     "environmental_protection": 1,
     "flags": [ "OVERSIZE", "OUTER", "WATERPROOF", "ALLOWS_NATURAL_ATTACKS" ]
+  },
+  {
+    "id": "shield_kite",
```
Comment (Rod995) [inline data/json/items/armor/misc.json:272]: That version of the kite shield that was [imported](https://github.com/CleverRaven/Cataclysm-DDA/pull/44879/files#diff-71279ee2b4d504e9901e283768c6094d463adaa5ae427e2ef057588daeaaaeee) is  outdated, I'd recommend importing from the mod author's [revised repository](https://github.com/chaosvolt/cdda_medieval_mod_reborn/blob/master/Medieval_Mod_Reborn/items/armor.json) instead. The updated version fixed some old oversights in the item's JSON not found in your imported version.

**Review by I-am-Erk — COMMENTED** [review]
I love this intensely and am ashamed we didn't have it earlier.

For the kite shield, I believe shields are awaiting proper implementation of blocking mechanics. May want to leave that out for now, perhaps you add a helmet with a severed head inside it instead?

File: data/json/items/armor/misc.json (hunk 1 above)
Comment (Night-Pryanik) [reply data/json/items/armor/misc.json:272]: Will update it soon, thanks.

**Comment by Night-Pryanik** [comment]
Well, I'm not tied to the shield at all, I can drop it entirely if it's causing any problems at the moment.

**Pull request merged**